contrib/make_clean

```diff
@@ -1,7 +1,21 @@
 #!/bin/bash
+# Delete everything in the repository that is not part of the git repository
+# except the release files and build artifacts we want to be cached.
 
 set -e
 
-here=$(dirname $(realpath "$0" 2> /dev/null || grealpath "$0"))
+if [ ! -f 'electrum-abc' ]; then
+    fail "Please run this script from the top-level Electrum ABC git directory"
+fi
 
-rm -Rf "$here/build"
+source $( dirname -- "$BASH_SOURCE" )/base.sh
+
+git clean -xdff \
+ --exclude=dist \
+ --exclude=packages \
+ --exclude=contrib/build \
+ --exclude=contrib/osx/.cache \
+ --exclude=electrumabc/locale \
+ --exclude=contrib/osx/*.dylib \
+ --exclude=electrumabc/*.dylib \
+ --exclude=electrumabc/*.so.0
```

Comment:
Excluding the build artifacts is a bad idea, this is the opposite to what a user will expect when running a clean script. If you want caching, pass an out of tree cache directory at build time.